node/peer.js

```diff
@@ -443,7 +443,7 @@ PreferOutgoingHandler.prototype.getScore = function getScore() {
     switch (qos) {
         case TIER_ONLY_INCOMING:
             if (!self.peer.channel.destroyed) {
-                self.peer.connect();
+                self.peer.connect(true);
```

Comment:
aye this is clearly a bug in prior code, since it would cause us to not initiate the new connection on the original node which the service connected to